Data/Aeson.hs

```diff
@@ -220,6 +224,15 @@ eitherDecodeStrict' =
   eitherFormatError . eitherDecodeStrictWith jsonEOF' ifromJSON
 {-# INLINE eitherDecodeStrict' #-}
 
+-- | Decode a nested JSON-encoded string.
+embeddedJSON :: (FromJSON a) => String -> (Value -> Parser a)
+embeddedJSON str = withText str (either fail return . eitherDecode . fromStrict . T.encodeUtf8)
+  where
+#if MIN_VERSION_bytestring(0, 9, 2)
+  fromStrict = L.fromChunks . (:[])
+#else
+  fromStrict = L.fromStrict
+#endif
 -- $use
```

Comment:
I guess there should also be an INLINE pragma like the rest of the functions have, since the implementation is just composition it should be fine to do.

Reply:
Will do.